cql3/expr/expression.cc

```diff
@@ -411,10 +404,11 @@ bool is_one_of(const column_value& col, term& rhs, const column_value_eval_bag&
             });
     } else if (auto mkr = dynamic_cast<lists::marker*>(&rhs)) {
         // This is `a IN ?`.  RHS elements are values representable as bytes_opt.
```

Comment:
Done, but it required some additional code.
A list of `IN` values, can contain a `NULL`, which is a problem because a normal list can't contain any `NULLs`.
I solved this by adding `evaluate_IN_list`, which behaves like `evaluate`, but ignores any `NULL` values in the list.
We can ignore any `NULLs` in an `IN` list, because nothing can be equal to `NULL`.

I also found a bug where I set the wrong type of a `list` constant, it's fixed now.